runtime/memory.c

```diff
@@ -673,12 +676,29 @@ CAMLexport caml_stat_block caml_stat_resize_noexc(caml_stat_block b, asize_t sz)
        while other domains access the pool concurrently. */
     unlink_pool_block(pb);
     /* Reallocating */
-    pb_new = realloc(pb, sz + SIZEOF_POOL_BLOCK);
+#ifdef _WIN32
+    pb_new = _aligned_realloc(pb, sizeof(struct pool_block) + sz,
+                              pool_block_align);
+#else
+#  if defined(__APPLE__)
+    size_t old_size = malloc_size(pb);
+#  else
+    size_t old_size = malloc_usable_size(pb);
```

Comment:
Won't work on non-glibc non-jemalloc systems.

Also, quoting jemalloc documentation:
```  The malloc_usable_size()
     function is not a mechanism for in-place realloc(); rather it is provided
     solely as a tool for introspection purposes. Any discrepancy between the
     requested allocation size and the size reported by malloc_usable_size()
     should not be depended on, since such behavior is entirely
     implementation-dependent.
```